### PR TITLE
proxygen: use `free_port`

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This should make testing this more reliable on large PRs. See #181332.

### CODE REVIEW

**Comment by carlocab** [comment]
Strict linkage failures can be handled separately to help unblock https://github.com/Homebrew/homebrew-core/pull/181332.

File: Formula/p/proxygen.rb [hunk 1]
```diff
@@ -43,9 +43,10 @@ def install
   end
 
   test do
-    pid = spawn bin/"proxygen_echo"
+    port = free_port
+    pid = spawn(bin/"proxygen_echo", "--http_port", port.to_s)
     sleep 10
```
Comment (cho-m) [inline Formula/p/proxygen.rb:48]: I am thinking failure may be timing related given only happening on Intel runners, which seem to getter slower.

Comment (cho-m) [reply Formula/p/proxygen.rb]: Doubling sleep to see if it helps. May be worth having some logic to increase any `sleep` for test retries as common reason retry passes is timing.

**Review by cho-m — APPROVED** [review]

**Comment by cho-m** [comment]
Still having issues after longer timeout. Can't tell if Monterey is just slow or some other issue from logs.

**Review by branchv — APPROVED** [review]

**Pull request merged**